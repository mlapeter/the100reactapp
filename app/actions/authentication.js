export const FETCH_TOKEN = "FETCH_TOKEN";
export const FETCH_TOKEN_RESULT = "FETCH_TOKEN_RESULT";
export const FETCH_TOKEN_ERROR = "FETCH_TOKEN_ERROR";

export const DECODE_TOKEN = "DECODE_TOKEN";
export const DECODE_TOKEN_RESULT = "DECODE_TOKEN_RESULT";
export const DECODE_TOKEN_ERROR = "DECODE_TOKEN_ERROR";
export const REMOVE_TOKEN = "REMOVE_TOKEN";
export const REMOVE_TOKEN_ERROR = "REMOVE_TOKEN_ERROR";
export const SET_FIREBASE_TOKEN = "SET_FIREBASE_TOKEN";
export const RESET_PASSWORD = "RESET_PASSWORD";
export const RESET_PASSWORD_RESULT = "RESET_PASSWORD_RESULT";
export const RESET_PASSWORD_ERROR = "RESET_PASSWORD_ERROR";

export const fetchToken = (username, password, temp_auth_token) => ({
  type: FETCH_TOKEN,
  username,
  password,
  temp_auth_token
});

export const decodeToken = token => ({
  type: DECODE_TOKEN,
  token
});

export const removeToken = () => ({
  type: REMOVE_TOKEN
});

export const setFirebaseToken = token => ({
  type: SET_FIREBASE_TOKEN,
  token
});

export const resetPassword = resetPasswordEmail => ({
  type: RESET_PASSWORD,
  resetPasswordEmail
});
