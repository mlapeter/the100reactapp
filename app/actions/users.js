export const UPDATE_USER = "UPDATE_USER";
export const UPDATE_USER_RESULT = "UPDATE_USER_RESULT";
export const UPDATE_USER_ERROR = "UPDATE_USER_ERROR";

export const FETCH_USER = "FETCH_USER";
export const FETCH_USER_RESULT = "FETCH_USER_RESULT";
export const FETCH_USER_ERROR = "FETCH_USER_ERROR";

export const FETCH_CURRENT_USER = "FETCH_CURRENT_USER";
export const FETCH_CURRENT_USER_RESULT = "FETCH_CURRENT_USER_RESULT";
export const CLEAR_CURRENT_USER = "CLEAR_CURRENT_USER";

export const FETCH_FRIENDS = "FETCH_FRIENDS";
export const FETCH_FRIENDS_RESULT = "FETCH_FRIENDS_RESULT";
export const FETCH_FRIENDS_ERROR = "FETCH_FRIENDS_ERROR";
export const FETCH_FRIENDS_NO_DATA = "FETCH_FRIENDS_NO_DATA";
export const LOAD_MORE_FRIENDS = "LOAD_MORE_FRIENDS";
export const LOAD_MORE_FRIENDS_RESULT = "LOAD_MORE_FRIENDS_RESULT";
export const CHANGE_FRIENDS_PAGE = "CHANGE_FRIENDS_PAGE";
export const REFRESH_FRIENDS = "REFRESH_FRIENDS";

export const FETCH_PENDING_FRIENDS = "FETCH_PENDING_FRIENDS";
export const FETCH_PENDING_FRIENDS_RESULT = "FETCH_PENDING_FRIENDS_RESULT";
export const FETCH_PENDING_FRIENDS_ERROR = "FETCH_PENDING_FRIENDS_ERROR";
export const FETCH_PENDING_FRIENDS_NO_DATA = "FETCH_PENDING_FRIENDS_NO_DATA";
export const LOAD_MORE_PENDING_FRIENDS = "LOAD_MORE_PENDING_FRIENDS";
export const LOAD_MORE_PENDING_FRIENDS_RESULT =
  "LOAD_MORE_PENDING_FRIENDS_RESULT";

export const CHANGE_PENDING_FRIENDS_PAGE = "CHANGE_PENDING_FRIENDS_PAGE";
export const REFRESH_PENDING_FRIENDS = "REFRESH_PENDING_FRIENDS";

export const FETCH_GROUP_MEMBERS = "FETCH_GROUP_MEMBERS";
export const FETCH_GROUP_MEMBERS_RESULT = "FETCH_GROUP_MEMBERS_RESULT";
export const FETCH_GROUP_MEMBERS_ERROR = "FETCH_GROUP_MEMBERS_ERROR";
export const FETCH_GROUP_MEMBERS_NO_DATA = "FETCH_GROUP_MEMBERS_NO_DATA";
export const LOAD_MORE_GROUP_MEMBERS = "LOAD_MORE_GROUP_MEMBERS";
export const LOAD_MORE_GROUP_MEMBERS_RESULT = "LOAD_MORE_GROUP_MEMBERS_RESULT";
export const CHANGE_GROUP_MEMBERS_PAGE = "CHANGE_GROUP_MEMBERS_PAGE";
export const REFRESH_GROUP_MEMBERS = "REFRESH_GROUP_MEMBERS";

export const fetchUser = userId => ({
  type: FETCH_USER,
  userId
});

export const fetchCurrentUser = () => ({
  type: FETCH_CURRENT_USER
});

export const updateUser = currentUser => ({
  type: UPDATE_USER,
  currentUser
});

export const clearCurrentUser = () => ({
  type: CLEAR_CURRENT_USER
});

export const fetchFriends = () => ({
  type: FETCH_FRIENDS
});

export const loadMoreFriends = () => ({
  type: LOAD_MORE_FRIENDS
});

export const changeFriendsPage = () => ({
  type: CHANGE_FRIENDS_PAGE
});

export const refreshFriends = () => ({
  type: REFRESH_FRIENDS
});

export const fetchPendingFriends = () => ({
  type: FETCH_PENDING_FRIENDS
});

export const loadMorePendingFriends = () => ({
  type: LOAD_MORE_PENDING_FRIENDS
});

export const refreshPendingFriends = () => ({
  type: REFRESH_PENDING_FRIENDS
});

export const fetchGroupMembers = () => ({
  type: FETCH_GROUP_MEMBERS
});

export const loadMoreGroupMembers = () => ({
  type: LOAD_MORE_GROUP_MEMBERS
});

export const refreshGroupMembers = () => ({
  type: REFRESH_GROUP_MEMBERS
});
