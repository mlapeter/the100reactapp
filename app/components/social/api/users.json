[
    {
        "id": "derek.russel",
        "name": "Derek Russel",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/8bbac5ad06b4a569b8a446825f7371c81ebac821.png?alt=media&token=73b33332-c587-464b-af68-52554221b73a",
        "caption": ""
    },
    {
        "id": "jmitch",
        "name": "Joe Mitchell",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/9b5cb1a55d786aebebfe7379dc16b8ccf0e81942.png?alt=media&token=6f3c5b07-1b67-478b-9b46-15cfee8d6f54",
        "caption": ""
    },
    {
        "id": "monicaa",
        "name": "Monica Dixon",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/a71de5589d604ece4a685c2c270267cebe192be8.png?alt=media&token=0d91856a-8a43-4b58-8e44-57becc3f34eb",
        "caption": ""
    },
    {
        "id": "andrea.schmidt",
        "name": "Andrea Schmidt",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/6e37e1c924aa55072f25e47ede573285a6fb69bf.png?alt=media&token=ae4e9186-91b8-4748-963a-02138a3e395c",
        "caption": ""
    },
    {
        "id": "alexandergarcia",
        "name": "Alex Garcia",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/a2c02707b8423da62b591e7111e790a77917cf14.png?alt=media&token=98e89375-36ce-4477-b19a-7d68194de3e8",
        "caption": ""
    },
    {
        "id": "mjcole",
        "name": "Michael Cole",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/40eabfbf1c4cd057304d9a18e260d42c7025b5ca.png?alt=media&token=ebbc0a41-61db-4bed-a0c5-8f088eb16c21",
        "caption": ""
    },
    {
        "id": "schavez",
        "name": "Sylvia Chavez",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/6e37e1c924aa55072f25e47ede573285a6fb69bf.png?alt=media&token=ae4e9186-91b8-4748-963a-02138a3e395c",
        "caption": "Aspiring creative writer. I love ðŸŒ¶ food and good people.",
        "cover": {
            "uri":  "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/e88d0539c04f38d3b7f7d030779f11feb26c3660.png?alt=media&token=7d2041fa-ed1f-41ee-ad6a-8af1531d657b",
            "preview": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAICAYAAADA+m62AAAAAXNSR0IArs4c6QAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAACXBIWXMAAAsTAAALEwEAmpwYAAABWWlUWHRYTUw6Y29tLmFkb2JlLnhtcAAAAAAAPHg6eG1wbWV0YSB4bWxuczp4PSJhZG9iZTpuczptZXRhLyIgeDp4bXB0az0iWE1QIENvcmUgNS40LjAiPgogICA8cmRmOlJERiB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiPgogICAgICA8cmRmOkRlc2NyaXB0aW9uIHJkZjphYm91dD0iIgogICAgICAgICAgICB4bWxuczp0aWZmPSJodHRwOi8vbnMuYWRvYmUuY29tL3RpZmYvMS4wLyI+CiAgICAgICAgIDx0aWZmOk9yaWVudGF0aW9uPjE8L3RpZmY6T3JpZW50YXRpb24+CiAgICAgIDwvcmRmOkRlc2NyaXB0aW9uPgogICA8L3JkZjpSREY+CjwveDp4bXBtZXRhPgpMwidZAAAA8ElEQVQYGRWPS1LCQABEX2YmwyR8EmRheRHv4Vm8rUstsQoKMAbI/Mew7a5+3V29vr+VKxa5KmSRmI6J4gSls6QQSHvB/X5H9M8Na9MQfuB8vlG2Gd0I5K+EKpONx/95VJSebqtnsWC/J1x50CFOkTBm4tpDn1Ff+4FWStKiUK0q4ikwVhEnA3FISJ9JtUe8mJ48FcIlUhfY7TYsQv2oousNEoWUGuWspVE1q7wgztRjHphyoNdLyrzoGiY27RJxGE/E+Z3QEpcd++FI6yW6NVxxSAvqpmZ/junaYHXg4/RJf1Y8dVucCUyHEWMNfj72D6rqgcOmcLiwAAAAAElFTkSuQmCC"
        }
    },
    {
        "id": "griffinsof",
        "name": "Sofia Griffin",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/8d8edc4c628969721a3127824566bcf7bef353c9.png?alt=media&token=99e98755-78ec-439b-ac51-778cba2d2cda",
        "caption": ""
    },
    {
        "id": "itsmesuri",
        "name": "Suri H.",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/881c9bfb4e5d3578355f328be01fe1d5152bbb27.png?alt=media&token=5ebcebcd-0c90-4d9d-8c45-daa05e6f1f41",
        "caption": ""
    }
]
