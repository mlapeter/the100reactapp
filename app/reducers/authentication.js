import {
  FETCH_TOKEN,
  FETCH_TOKEN_RESULT,
  FETCH_TOKEN_ERROR,
  DECODE_TOKEN,
  DECODE_TOKEN_RESULT,
  DECODE_TOKEN_ERROR,
  REMOVE_TOKEN,
  REMOVE_TOKEN_ERROR,
  SET_FIREBASE_TOKEN
} from "../actions/authentication";

const initialState = {
  appLoading: true,
  isLoading: false,
  isAuthed: false,
  user: {},
  token: "",
  firebaseToken: ""
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TOKEN:
      return {
        ...state,
        username: action.username,
        password: action.password,
        isLoading: true
      };
    case FETCH_TOKEN_RESULT:
      return {
        ...state,
        token: action.token,
        firebaseToken: action.firebaseToken
      };
    case FETCH_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        token: "",
        firebaseToken: ""
      };
    case DECODE_TOKEN:
      return {
        ...state,
        token: action.token,
        isLoading: true
      };
    case DECODE_TOKEN_RESULT:
      return {
        ...state,
        user: action.result,
        isLoading: false,
        isAuthed: true
      };
    case DECODE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false
      };
    case REMOVE_TOKEN:
      return {
        ...state,
        isAuthed: false,
        isLoading: false,
        user: {},
        token: "",
        firebaseToken: ""
      };
    case REMOVE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        user: {},
        token: ""
      };
    case SET_FIREBASE_TOKEN:
      return {
        ...state,
        firebaseToken: action.token
      };
    default:
      return state;
  }
};
