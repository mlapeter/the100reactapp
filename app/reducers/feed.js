import {
  FETCH_FEED,
  FETCH_FEED_RESULT,
  FETCH_FEED_ERROR,
  FETCH_FEED_NO_DATA,
  LOAD_MORE_FEED_ITEMS,
  LOAD_MORE_FEED_ITEMS_RESULT,
  CHANGE_FEED_PAGE
} from "../actions/feed";

const initialState = {
  isLoading: false,
  feedItems: [],
  feedPage: 1,
  moreFeedItemsAvailable: true
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_FEED:
      return {
        ...state,
        isLoading: true,
        moreFeedItemsAvailable: true
      };
    case FETCH_FEED_RESULT:
      return {
        ...state,
        feedItems: action.result,
        isLoading: false
      };
    case FETCH_FEED_ERROR:
      return {
        ...state,
        error: action.error,
        isLoading: false
      };
    case FETCH_FEED_NO_DATA:
      return {
        ...state,
        isLoading: false,
        moreFeedItemsAvailable: false
      };
    case LOAD_MORE_FEED_ITEMS:
      return {
        ...state,
        isLoading: true
      };
    case LOAD_MORE_FEED_ITEMS_RESULT:
      return {
        ...state,
        feedItems: [...state.feedItems, ...action.result],
        isLoading: false
      };

    case CHANGE_FEED_PAGE:
      return {
        ...state,
        feedPage: action.page
      };
    default:
      return state;
  }
};
