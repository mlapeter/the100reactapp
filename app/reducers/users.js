import {
  UPDATE_USER,
  UPDATE_USER_RESULT,
  UPDATE_USER_ERROR,
  UPDATE_USER_PUSH_TOKEN,
  FETCH_USER,
  FETCH_USER_RESULT,
  FETCH_USER_ERROR,
  FETCH_CURRENT_USER,
  FETCH_CURRENT_USER_RESULT,
  CLEAR_CURRENT_USER,
  FETCH_FRIENDS,
  FETCH_FRIENDS_RESULT,
  FETCH_FRIENDS_ERROR,
  FETCH_FRIENDS_NO_DATA,
  LOAD_MORE_FRIENDS,
  LOAD_MORE_FRIENDS_RESULT,
  CHANGE_FRIENDS_PAGE,
  FETCH_PENDING_FRIENDS,
  FETCH_PENDING_FRIENDS_RESULT,
  FETCH_PENDING_FRIENDS_ERROR,
  FETCH_PENDING_FRIENDS_NO_DATA,
  LOAD_MORE_PENDING_FRIENDS,
  LOAD_MORE_PENDING_FRIENDS_RESULT,
  CHANGE_PENDING_FRIENDS_PAGE,
  FETCH_GROUP_MEMBERS,
  FETCH_GROUP_MEMBERS_RESULT,
  FETCH_GROUP_MEMBERS_ERROR,
  FETCH_GROUP_MEMBERS_NO_DATA,
  LOAD_MORE_GROUP_MEMBERS,
  LOAD_MORE_GROUP_MEMBERS_RESULT,
  CHANGE_GROUP_MEMBERS_PAGE
} from "../actions/users";

const initialState = {
  isUpdating: false,
  userLoading: true,
  user: {},
  currentUser: null,
  friends: [],
  groupMembers: [],
  pendingFriends: [],
  isLoadingFriends: false,
  isLoadingGroupMembers: false,
  isLoadingPendingFriends: false,
  moreFriendsAvailable: true,
  moreGroupMembersAvailable: true,
  morePendingFriendsAvailable: true,
  friendsPage: 1,
  groupMembersPage: 1,
  pendingFriendsPage: 1,
  expoPushToken: null
};

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_USER:
      return {
        ...state,
        isUpdating: true,
        currentUser: action.currentUser
      };
    case UPDATE_USER_RESULT:
      return {
        ...state,
        isUpdating: false,
        success: true,
        successAt: new Date()
        // currentUser: { ...state.currentUser, ...action.currentUser }
      };
    case UPDATE_USER_ERROR:
      return {
        ...state,
        isUpdating: false,
        error: action.error,
        errorAt: new Date()
      };
    case UPDATE_USER_PUSH_TOKEN:
      return {
        ...state,
        expoPushToken: action.token
      };

    case FETCH_USER:
      return {
        ...state,
        userId: action.userId,
        userLoading: true
      };
    case FETCH_USER_RESULT:
      return {
        ...state,
        user: action.result,
        userLoading: false
      };
    case FETCH_USER_ERROR:
      return {
        ...state,
        user: {},
        error: action.error,
        errorAt: new Date(),
        userLoading: false,
        currentUserLoading: false
      };
    case FETCH_CURRENT_USER:
      return {
        ...state,
        currentUserLoading: true
      };
    case FETCH_CURRENT_USER_RESULT:
      return {
        ...state,
        currentUser: action.result,
        currentUserLoading: false
      };
    case CLEAR_CURRENT_USER:
      return {
        ...state,
        user: {},
        currentUser: null,
        friends: [],
        groupMembers: [],
        pendingFriends: [],
        currentUserLoading: false
      };
    case FETCH_FRIENDS:
      return {
        ...state,
        moreFriendsAvailable: true,
        isLoadingFriends: true
      };
    case FETCH_FRIENDS_RESULT:
      return {
        ...state,
        moreFriendsAvailable: true,
        isLoadingFriends: false,
        friends: action.result
      };
    case FETCH_FRIENDS_ERROR:
      return {
        ...state,
        moreFriendsAvailable: true,
        isLoadingFriends: false,
        error: action.error
      };
    case FETCH_FRIENDS_NO_DATA:
      return {
        ...state,
        moreFriendsAvailable: false,
        isLoadingFriends: false
      };
    case LOAD_MORE_FRIENDS:
      return {
        ...state,
        isLoadingFriends: true
      };
    case LOAD_MORE_FRIENDS_RESULT:
      return {
        ...state,
        friends: [...state.friends, ...action.result],
        isLoadingFriends: false
      };
    case CHANGE_FRIENDS_PAGE:
      return {
        ...state,
        friendsPage: action.page
      };
    case FETCH_GROUP_MEMBERS:
      return {
        ...state,
        moreGroupMembersAvailable: true,
        isLoadingGroupMembers: true
      };
    case FETCH_GROUP_MEMBERS_RESULT:
      return {
        ...state,
        moreGroupMembersAvailable: true,
        isLoadingGroupMembers: false,
        groupMembers: action.result
      };
    case FETCH_GROUP_MEMBERS_ERROR:
      return {
        ...state,
        moreGroupMembersAvailable: true,
        isLoadingGroupMembers: false,
        error: action.error
      };
    case FETCH_GROUP_MEMBERS_NO_DATA:
      return {
        ...state,
        moreGroupMembersAvailable: false,
        isLoadingGroupMembers: false
      };
    case LOAD_MORE_GROUP_MEMBERS:
      return {
        ...state,
        isLoadingGroupMembers: true
      };
    case LOAD_MORE_GROUP_MEMBERS_RESULT:
      return {
        ...state,
        groupMembers: [...state.groupMembers, ...action.result],
        isLoadingGroupMembers: false
      };
    case CHANGE_GROUP_MEMBERS_PAGE:
      return {
        ...state,
        groupMembersPage: action.page
      };
    case FETCH_PENDING_FRIENDS:
      return {
        ...state,
        morePendingFriendsAvailable: true,
        isLoadingPendingFriends: true
      };
    case FETCH_PENDING_FRIENDS_RESULT:
      return {
        ...state,
        morePendingFriendsAvailable: true,
        isLoadingPendingFriends: false,
        pendingFriends: action.result
      };
    case FETCH_PENDING_FRIENDS_ERROR:
      return {
        ...state,
        morePendingFriendsAvailable: true,
        isLoadingPendingFriends: false,
        error: action.error
      };
    case FETCH_PENDING_FRIENDS_NO_DATA:
      return {
        ...state,
        morePendingFriendsAvailable: false,
        isLoadingPendingFriends: false
      };
    case LOAD_MORE_PENDING_FRIENDS:
      return {
        ...state,
        isLoadingPendingFriends: true
      };
    case LOAD_MORE_PENDING_FRIENDS_RESULT:
      return {
        ...state,
        pendingFriends: [...state.pendingFriends, ...action.result],
        isLoadingPendingFriends: false
      };
    case CHANGE_PENDING_FRIENDS_PAGE:
      return {
        ...state,
        pendingFriendsPage: action.page
      };

    default:
      return state;
  }
};
