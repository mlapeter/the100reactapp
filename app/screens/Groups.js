// @flow
import * as React from "react";
import { Alert, Image, StyleSheet, View } from "react-native";
import { connectAlert } from "../components/Alert";
import { connect } from "react-redux";
import { colors, fontSizes, fontStyles, styleSheet } from "../styles";
import TopNav from "../components/TopNav/TopNav";
import GroupsList from "../components/GroupsList";
import {
  type NavigationNavigatorProps,
  type NavigationScreenProp
} from "react-navigation";

type Group = {
  id: number,
  name: string,
  header_background_image_api: string
};

type User = {
  id: number
};

type GroupsProps = {
  groups: Array<Group>,
  user: User,
  isLoading: boolean,
  navigation: NavigationScreenProp<{}>
};

class Groups extends React.Component<GroupsProps> {
  render() {
    return (
      <View style={styles.container}>
        <TopNav
          title="Groups"
          user={this.props.user}
          navigation={this.props.navigation}
        />

        <GroupsList
          groups={this.props.groups}
          navigation={this.props.navigation}
          isLoading={this.props.isLoading}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingBottom: styleSheet.spacing.small,
    backgroundColor: colors.lightGray
  }
});

const mapStateToProps = state => {
  const groups = state.users.currentUser.groups;
  const isLoading = state.group.isLoading;
  const user = state.authentication.user;

  return {
    groups,
    isLoading,
    groupsError: state.group.error,
    user
  };
};

export default connect(mapStateToProps)(connectAlert(Groups));
