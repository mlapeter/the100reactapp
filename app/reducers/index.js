import { combineReducers } from "redux";

import authentication from "./authentication";
import search from "./search";
import notifications from "./notifications";
import users from "./users";
import group from "./group";
import gamingSessions from "./gamingSessions";
import onboarding from "./onboarding";
import conversations from "./conversations";
import feed from "./feed";

export default combineReducers({
  authentication,
  search,
  notifications,
  users,
  group,
  gamingSessions,
  onboarding,
  conversations,
  feed
});
