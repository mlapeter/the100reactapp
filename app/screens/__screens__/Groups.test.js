import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import Groups from "../Groups";
import { initialState as authenticationInitialState } from "../../reducers/authentication";
import { initialState as usersInitialState } from "../../reducers/users";

const mockStore = configureStore([]);

it("successfully renders", () => {
  const navigation = { state: { params: {} } };
  const initialState = {
    authentication: authenticationInitialState,
    users: usersInitialState
  };
  const rendered = shallow(<Groups navigation={navigation} />, {
    context: { store: mockStore(initialState) }
  });
  expect(rendered.dive().dive()).toMatchSnapshot();
});
