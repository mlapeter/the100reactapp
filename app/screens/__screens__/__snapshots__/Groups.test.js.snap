// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`successfully renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Groups
    dispatch={[Function]}
    navigation={
      Object {
        "state": Object {
          "params": Object {},
        },
      }
    }
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TopNav
          navigation={
            Object {
              "state": Object {
                "params": Object {},
              },
            }
          }
          title="Groups"
          user={Object {}}
        />,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          You haven't joined any groups yet! You can join any group you like by tapping the join icon at the top right of the group page.
        </Text>,
      ],
      "style": Object {
        "backgroundColor": "#F3F3F3",
        "flex": 1,
        "paddingBottom": 16,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "navigation": Object {
            "state": Object {
              "params": Object {},
            },
          },
          "title": "Groups",
          "user": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "You haven't joined any groups yet! You can join any group you like by tapping the join icon at the top right of the group page.",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "You haven't joined any groups yet! You can join any group you like by tapping the join icon at the top right of the group page.",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TopNav
            navigation={
              Object {
                "state": Object {
                  "params": Object {},
                },
              }
            }
            title="Groups"
            user={Object {}}
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            You haven't joined any groups yet! You can join any group you like by tapping the join icon at the top right of the group page.
          </Text>,
        ],
        "style": Object {
          "backgroundColor": "#F3F3F3",
          "flex": 1,
          "paddingBottom": 16,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "navigation": Object {
              "state": Object {
                "params": Object {},
              },
            },
            "title": "Groups",
            "user": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "You haven't joined any groups yet! You can join any group you like by tapping the join icon at the top right of the group page.",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "You haven't joined any groups yet! You can join any group you like by tapping the join icon at the top right of the group page.",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
